src/en/sunshinebutterflyscans/src/eu/kanade/tachiyomi/extension/en/sunshinebutterflyscans/SunshineButterflyScans.kt

```diff
@@ -1,5 +1,244 @@
+        add("Host", baseUrl.toHttpUrl().host)
+    }
+
+    private val apiHeaders by lazy { apiHeadersBuilder().build() }
+
+    private val json: Json by injectLazy()
+
+    private val chaptersData by lazy {
+        client.newCall(
+            GET("$baseUrl/json/chapters.json", apiHeaders),
+        ).execute().parseAs<List<Dto>>().groupBy {
+            it.series
+        }.values.map { it.sortedByDescending { it.num } }
+    }
+
+    // ============================== Popular ===============================
+
+    override fun fetchPopularManga(page: Int): Observable<MangasPage> {
+        val mangaList = chaptersData.sortedBy {
+            it.first().series
+        }.map {
+            it.first().toSManga(cdnUrl)
+        }
+
+        return Observable.just(MangasPage(mangaList, false))
+    }
+
+    override fun popularMangaRequest(page: Int): Request =
+        throw UnsupportedOperationException()
+
+    override fun popularMangaParse(response: Response): MangasPage =
+        throw UnsupportedOperationException()
+
+    // =============================== Latest ===============================
+
+    override fun fetchLatestUpdates(page: Int): Observable<MangasPage> {
+        val mangaList = chaptersData.sortedByDescending {
+            it.first().timestamp.toLongOrNull() ?: Long.MAX_VALUE
+        }.map {
+            it.first().toSManga(cdnUrl)
+        }
+
+        return Observable.just(MangasPage(mangaList, false))
+    }
+
+    override fun latestUpdatesRequest(page: Int): Request =
+        throw UnsupportedOperationException()
+
+    override fun latestUpdatesParse(response: Response): MangasPage =
+        throw UnsupportedOperationException()
+
+    // =============================== Search ===============================
+
+    override fun fetchSearchManga(page: Int, query: String, filters: FilterList): Observable<MangasPage> {
+        val selectedStatus = filters.filterIsInstance<StatusFilter>().first().toUriPart()
+
+        val predicate: (List<Dto>) -> Boolean = { chapterList ->
+            if (query.isNotEmpty()) {
+                chapterList.first().series.contains(query, true)
+            } else {
+                chapterList.first().projectStatus.contains(selectedStatus)
+            }
+        }
+
+        val mangaList = chaptersData.sortedBy {
+            it.first().series
+        }.filter(predicate).map {
+            it.first().toSManga(cdnUrl)
+        }
+
+        return Observable.just(MangasPage(mangaList, false))
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request =
+        throw UnsupportedOperationException()
+
+    override fun searchMangaParse(response: Response): MangasPage =
+        throw UnsupportedOperationException()
+
+    // =============================== Filters ==============================
+
+    override fun getFilterList(): FilterList = FilterList(
+        Filter.Header("Text search ignores filters"),
```

Comment:
since filter is client side, you can techincally make them both work at same time; unless this a delibrate decision?

Reply:
I made it so it works like the site, but I can make it work with both sure